packages/core/src/shared/telemetry/util.ts

```diff
@@ -90,6 +91,37 @@ export function convertLegacy(value: unknown): boolean {
     }
 }
 
+/**
+ * Returns an identifier that uniquely identifies a single application
+ * instance/window of a specific IDE. I.e if I have multiple VS Code
+ * windows open each one will have a unique session ID. This session ID
+ * can be used in conjunction with the client ID to differntiate between
+ * different VS Code windows on a users machine.
+ *
+ * ### Rules:
+ *
+ * - On startup of a new application instance, a new session ID must be created.
+ *   - This identifier must be a `UUID`, it is enforced by the telemetry service.
+ * - The session ID must be different from all other IDs on the machine
+ * - A session ID exists until the application instance is terminated.
```

Comment:
```suggestion
 * - A session ID exists until the Toolkit extension instance is terminated.
```

Reply:
If I have Amazon Q and Toolkit on my VS Code window, and Toolkit crashes, I want the session ID to remain the same for Amazon Q.

I'm not sure if this suggested change gets that across